### ISSUE TITLE
Domain Forwarding: Support multiple redirects

### ISSUE THREAD

**Issue by paulopmt1** [Automattic/wp-calypso#81290]
PT and extra context: pb5gDS-3ig-p2
Design: pb5gDS-3ig-p2#design

After we implement [the new Domain Forwarding block](https://github.com/orgs/Automattic/projects/510/views/1?pane=issue&itemId=37265150) we can support multiple redirects:

![Image](https://user-images.githubusercontent.com/1044309/264412678-d20e7f46-a782-4140-ba21-90964480c872.png)

### Listing status
We want to use less space while listing Forwardings. More context: p1693859782346289-slack-CKZHG0QCR

![image](https://github.com/Automattic/wp-calypso/assets/1044309/23ae9df8-9f71-4eb9-b573-a3e8ed8e6236)

### Acceptance criteria:

- [x] User can add one domain and N subdomain redirects
- [x] Don't allow multiple domain redirects. We expect to handle that with an error message.
- [x] One redirect module should be open by default and be a Domain redirect.
- [ ] If a user deletes a redirect we will show them a modal saying: Are you sure you want to remove the redirect? CTA: Yes / Not now
- [x] Keep domain dropdown on editing a domain, but hide it for subdomain editions in case we already have one domain forwarding registered
- [x] Once we have the domain setup, we'll remove the domain selection on all but the domain
- [x] We will display remove forward button only after clicking editing
- [x] Keep the order by registration (domain_redirect_id ASC)
- [x] The Domain block listing will be yellow
- [x] Add forward blue and underline
- [x] Advanced settings should have the same Destination URL style (typography, color, and size)

**Comment by paulopmt1** [comment]
API changes: D120632-code

**Comment by paulopmt1** [comment]
From the previous chat with @JanaMW27:

- Keep domain dropdown on editing  
- Once we have the domain setup, we'll remove the domain selection on all but the domain
- We will display `remove forward` button only after clicking editing
- Keep the order by registration (domain_redirect_id ASC)
- The Domain block listing will be yellow
- Add forward blue and underline
- Advanced settings should have the same Destination URL style (typography, color, and size)
